src/components/calculations/CalculationSelect.js

```diff
@@ -0,0 +1,52 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { SelectField } from '../core/index.js'
+import { useUserSettings } from '../UserSettingsProvider.js'
+
+// Load all calculations
+const CALCULATIONS_QUERY = {
+    calculations: {
+        resource: 'expressionDimensionItems',
+        params: ({ nameProperty }) => ({
+            fields: ['id', `${nameProperty}~rename(name)`],
+            paging: false,
+        }),
+    },
+}
+
+const CalculationSelect = ({ calculation, className, errorText, onChange }) => {
+    const { nameProperty } = useUserSettings()
+    const { loading, error, data } = useDataQuery(CALCULATIONS_QUERY, {
+        variables: { nameProperty },
+    })
+
+    const items = data?.calculations.expressionDimensionItems
+    const value = calculation?.id
+
+    return (
+        <SelectField
+            label={i18n.t('Calculation')}
+            loading={loading}
+            items={items}
+            value={value}
+            onChange={(dataItem) => onChange(dataItem, 'calculation')}
+            className={className}
+            errorText={
+                error?.message || (!calculation && errorText ? errorText : null)
+            }
+            filterable={true}
+            dataTest="calculationselect"
```

Comment:
We can have a more relevant text when there are no results found for filter, `noMatchText={i18n.t('No calculations found`)}`

Reply:
Changed to "No calculations found"

![Screenshot 2023-08-31 at 16 07 05](https://github.com/dhis2/maps-app/assets/548708/8a09ea75-d6d9-4215-9049-d3cba7866755)